### PR TITLE
logging: Enable SWO pin for STM32 SoCs 

### LINKED ISSUE #34342: No output on SWO pin (STM32L4)
**Describe the bug**
I'm not quite sure if this is a bug, but by default the SWO pin is not enabled to output data. I thing the debugger could do this, but I feel like asking the question though.

As a workaround, I added this line:

```diff
diff --git a/subsys/logging/log_backend_swo.c b/subsys/logging/log_backend_swo.c
index 203896ccd0..8e2e3da5cc 100644
--- a/subsys/logging/log_backend_swo.c
+++ b/subsys/logging/log_backend_swo.c
@@ -98,6 +98,8 @@ static void log_backend_swo_init(void)
         ITM->TCR  = 0x0001000D;
         /* Enable stimulus port used by the logger */
         ITM->TER  = 1 << ITM_PORT_LOGGER;
+        /* Enable the SWO pin */
+        DBGMCU->CR |= 0x20;
 }

 static void log_backend_swo_panic(struct log_backend const *const backend)
```

**To Reproduce**
 - Enable SWO logging as backend
 - Log simple messages in loop
 - No output is present on pin

**Expected behavior**
SWO pin outputs the logging data

**Environment (please complete the following information):**
 - OS: ArchLinux
 - Toolchain Zephyr SDK
 - BlackMagicProbe (Bumpy)
 - Version 9f858ac1b6bcf7f9262906f6719d51d5be4404ac

**Additional context**
Testing SWO with STM32L4

### PR DESCRIPTION
**Pull request by andresmanelli**
Set TRACE_MODE to asynchronous and enable trace output pin.

Fixes #34342

Signed-off-by: Andrés Manelli <am@toroid.io>

### CODE REVIEW

**Review by mnkp — CHANGES_REQUESTED** [review]
The code is STM32 specific. It should be placed somewhere in `soc/arm/st_stm32/`. It's typical that the platform needs some special, platform specific steps, to enable SWO. E.g. in case of Atmel SAM they are placed in `soc/arm/atmel_sam/same70/soc_config.c`.

**Comment by andresmanelli** [comment]
> The code is STM32 specific. It should be placed somewhere in `soc/arm/st_stm32/`. It's typical that the platform needs some special, platform specific steps, to enable SWO. E.g. in case of Atmel SAM they are placed in `soc/arm/atmel_sam/same70/soc_config.c`.

Thanks for the review. I will place this in the `soc` directory.

**Review by mnkp — APPROVED** [review]
LGTM for the SWO part. I'm approving the PR to remove my previous NACK. However, it needs to be verified by the STM32 maintainers.

File: soc/arm/st_stm32/common/soc_config.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (erwango) [inline soc/arm/st_stm32/common/soc_config.c:2]: Please add a copyright.

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/common/soc_config.c (hunk 1 above)
Comment (andresmanelli) [reply soc/arm/st_stm32/common/soc_config.c:4]: Done.

**Review by erwango — APPROVED** [review]

**Pull request merged**